### PR TITLE
Create release workflow

### PR DESCRIPTION
**Pull request by Felixoid**
Changelog category (leave one):
- Build/Testing/Packaging Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add action for published releases 

Detailed description / Documentation draft:
When releases `v\d\d[.]\d+[.]\d+[.]\d+-(testing|prestable|stable|lts)` are published, the workflow will:
- Download binary packages from S3
- Upload them to artifactory
- Push them to release assets

### CODE REVIEW

**Comment by Felixoid** [comment]
The workflow to check it works

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: DocsReleaseChecks
```
Comment (alesapin) [inline .github/workflows/docs.yml:1]: Let's rename file too -- `docs_release.yml`

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,55 +1,32 @@
...
-    branches:
-      - master
-    paths:
-      - 'docs/**'
-      - 'website/**'
-      - 'benchmark/**'
-      - 'docker/**'
+  release:
+    types:
+    - published
```
Comment (alesapin) [inline .github/workflows/release.yml:9]: Works for both `pre-release` and `release`?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -1,55 +1,32 @@
...
-          sudo rm -fr $TEMP_PATH
+    - name: Check out repository code
+      uses: actions/checkout@v2
+    - name: Download packages and push to Artifactory
+      env:
+        JFROG_API_KEY: ${{ secrets.JFROG_KEY_API_PACKAGES }}
+        TEMP_PATH: ${{runner.temp}}/release_packages
+      run: |
+        rm -rf "$TEMP_PATH" && mkdir -p "$TEMP_PATH"
+        python3 ./tests/ci/push_to_artifactory.py --release "${{ github.ref }}" \
```
Comment (alesapin) [inline .github/workflows/release.yml:23]: Maybe let's make a repo copy? What if your script changes something in repo directory?

**Review by alesapin — COMMENTED** [review]
But we still need to do all other release-related staff:
1. We have to manually update several files in repository. Look at the recent commit from the old release script for tweak releases: https://github.com/ClickHouse/ClickHouse/pull/30978/commits/9b13386cd4e53886785856a96ff189063590e7be and https://github.com/ClickHouse/ClickHouse/pull/30978/commits/1a254b97e2b56275fe89416febc07a946007f13a.
2. For new patch release (21.11 -> 21.12) we also need to create new PR like https://github.com/ClickHouse/ClickHouse/pull/32419 and change something in master https://github.com/ClickHouse/ClickHouse/commit/f2ccddb7be65179154d9b80a962c060828934529.
3. Push new docker (`clickhouse/clickhouse-server` and `clickhouse/clickhouse-client`) images to dockerhub.
4. Some docs how to create release.

File: .github/workflows/release.yml (hunk 3 above)
Comment (Felixoid) [reply .github/workflows/release.yml:30]: done

**Comment by Felixoid** [comment]
The arm64 packages will be addressed in #33664

**Review by alesapin — APPROVED** [review]
But we will need to update docs_release workflow id in rerun lambda.

**Comment by Felixoid** [comment]
For history, now DocksReleaseCheck is:

```
    {
      "id": 14738810,
      "node_id": "W_kwDOA5dJV84A4OV6",
      "name": "DocsReleaseChecks",
      "path": ".github/workflows/release.yml",
      "state": "active",
      "created_at": "2021-10-30T08:21:55.000Z",
      "updated_at": "2022-01-17T23:04:40.000Z",
      "url": "https://api.github.com/repos/ClickHouse/ClickHouse/actions/workflows/14738810",
      "html_url": "https://github.com/ClickHouse/ClickHouse/blob/master/.github/workflows/release.yml",
      "badge_url": "https://github.com/ClickHouse/ClickHouse/workflows/DocsReleaseChecks/badge.svg"
    },
```

**Pull request merged**